### PR TITLE
Fix #4309 - disable RUBYLIB env and use --include or -I using cli

### LINKED ISSUE #4309: Expected debug messages missing when using CLI --verbose flag
Issue overview
--------------
When I get a user file that won't open because of a problem object, a common approach, if I can't resolve through the OS Application or SketchUp Plugin diagnostic script logs, is to set it as a test model for one of my measures.

What I have found is that when I run the measure through the OpenStudio CLI I get far less logging than in Ruby, even with verbose flag. 

## Current Behavior
```
dgoldwas:0:tests$ openstudio --verbose diagnostic_report_test.rb
I, [2021-04-28T09:03:39.111262 #89519]  INFO -- : CLI Parsed Inputs: ["--verbose"] "execute_ruby_script" ["diagnostic_report_test.rb"]
D, [2021-04-28T09:03:39.111344 #89519] DEBUG -- : Main arguments are ["--verbose"]
D, [2021-04-28T09:03:39.111356 #89519] DEBUG -- : Sub-command is execute_ruby_script
D, [2021-04-28T09:03:39.111395 #89519] DEBUG -- : Sub-arguments are ["diagnostic_report_test.rb"]
D, [2021-04-28T09:03:39.111421 #89519] DEBUG -- : Parsing main_args ["--verbose"]
D, [2021-04-28T09:03:39.331563 #89519] DEBUG -- : Adding dependency on jaro_winkler '~> 1.5.4'
.... many more lines of adding dependency
D, [2021-04-28T09:03:39.362870 #89519] DEBUG -- : Processing 46 activation requests
D, [2021-04-28T09:03:39.366106 #89519] DEBUG -- : Activating gem :/ruby/2.7.0/bundler/specifications/jaro_winkler-1.5.4.gemspec
.... many more lines of activating gems
D, [2021-04-28T09:03:39.370702 #89519] DEBUG -- : Invoking command class: ExecuteRubyScript ["diagnostic_report_test.rb"]
I, [2021-04-28T09:03:39.370721 #89519]  INFO -- : ExecuteRubyScript, sub_argv = ["diagnostic_report_test.rb"]
D, [2021-04-28T09:03:39.379073 #89519] DEBUG -- : ExecuteRubyScript command: ["diagnostic_report_test.rb"]
D, [2021-04-28T09:03:39.379175 #89519] DEBUG -- : Path for the file to run: /Users/dgoldwas/Documents/temp_measures/diagnostic_report/tests/diagnostic_report_test.rb
D, [2021-04-28T09:03:39.379216 #89519] DEBUG -- : ARGV: []
/Applications/OpenStudio-3.2.0-alpha/Ruby/openstudio.bundle: warning: already initialized constant OpenStudio::FloatFormat_fixed
.... many more lines of bundle warnings
Run options: --seed 43095

# Running:

F

Finished in 1.125740s, 0.8883 runs/s, 0.8883 assertions/s.

  1) Failure:
DiagnosticReportTest#test_good_argument_values [/Users/dgoldwas/Documents/temp_measures/diagnostic_report/tests/diagnostic_report_test.rb:23]:
Expected false to be truthy.

1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
```

## Expected Behavior

The expected behavior should have the content below that shows up when I run the test with my system Ruby

```
dgoldwas:0:tests$ ruby diagnostic_report_test.rb
Run options: --seed 50936

# Running:

[utilities.idf.IdfFile] <0> Cannot find object type 'OS:Coil:Heating:NaturalGas' in Idd. Placing data in Catchall object.
[utilities.idf.IdfFile] <0> Cannot find object type 'Catchall' in Idd. Placing data in Catchall object.
[utilities.idf.IdfFile] <0> Cannot find object type 'Catchall' in Idd. Placing data in Catchall object.
[utilities.idf.IdfFile] <0> Cannot find object type 'Catchall' in Idd. Placing data in Catchall object.
[utilities.idf.IdfFile] <0> Cannot find object type 'Catchall' in Idd. Placing data in Catchall object.
[utilities.idf.IdfFile] <0> Cannot find object type 'Catchall' in Idd. Placing data in Catchall object.
[utilities.idf.IdfFile] <0> Cannot find object type 'Catchall' in Idd. Placing data in Catchall object.
[utilities.idf.IdfFile] <0> Cannot find object type 'Catchall' in Idd. Placing data in Catchall object.
[utilities.idf.WorkspaceObject] <0> Object of type 'OS:Connection', points to an object named {d3bcc350-6fcc-48d2-b70e-18536a02797e} from field 1, but that object cannot be located.
[utilities.idf.WorkspaceObject] <0> Object of type 'OS:Connection', points to an object named {d3bcc350-6fcc-48d2-b70e-18536a02797e} from field 3, but that object cannot be located.
[openstudio.model.Model] <0> Creating GenericModelObject for IddObjectType 'Catchall'.
[openstudio.osversion.VersionTranslator] <1> Model with Version 3.2.0 IDD is not valid to draft strictness level.
[openstudio.osversion.VersionTranslator] <1> The collection is INVALID at strictness level 'Draft', because of the errors:
Object     level data error of type NoIdd             .
Error pertains to an object of type 'Catchall', named ''.
Additional information about the error type: input data definition is missing or empty.
F

Finished in 1.091747s, 0.9160 runs/s, 0.9160 assertions/s.

  1) Failure:
DiagnosticReportTest#test_good_argument_values [diagnostic_report_test.rb:23]:
Expected false to be truthy.

1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
```

## Steps to Reproduce

1. To reproduce break a model by making a class that doesn't exist. In my case object was of type `OS:Coil:Heating:NaturalGas` instead of `OS:Coil:Heating:Gas`
2. Then make a new measure with OS app and replace the test model with your broken model

## Possible Solution

the verbose flat is adding lots of logs before the measure is actually called but doesn't impact the logging during the measure run. Maybe the tests can be altered but ideally OpenStudio CLI can be altered without having to change all of our tests. Maybe something needs to be passed into runner?

One fix could be in the CLI or when running measure tests to allow the developer to explicitly choose loglevel.

## Details

These details are from @jmarrec 

Loglevels: Fatal = 2, Error =1, Warn = 0, Info = -1, Debug = -2, Trace = -3.

By default the C++ LogLevel = 0, so you do see these Warnings when using system ruby. (See LoggerSingleton: https://github.com/NREL/OpenStudio/blob/e603410ac3567d44e7660294f3b06a8a2d591118/src/utilities/core/Logger.cpp#L47-L62)

When you use the CLI though, LogLevel is set to Error (1) by default, so the warning is filtered out. https://github.com/NREL/OpenStudio/blob/e603410ac3567d44e7660294f3b06a8a2d591118/src/cli/openstudio_cli.rb#L56

That being said, openstudio --verbose should set it to Debug (-2), so you should see these warnings. https://github.com/NREL/OpenStudio/blob/e603410ac3567d44e7660294f3b06a8a2d591118/src/cli/openstudio_cli.rb#L739-L743

Here's an example of showing it working... I'm going to call evaluate with a X that is above the max limit (1.0 by default), and that issues a warn.

**Ruby: it's shown**

```
$ ruby -e "require 'openstudio'; m = OpenStudio::Model::Model.new; c = OpenStudio::Model::CurveLinear.new(m); c.evaluate(10)"
[openstudio.model.CurveLinear] <0> Supplied x is above the maximumValueofx, resetting it.
```

**CLI: it's not shown by default**

```
$ openstudio -e "m = OpenStudio::Model::Model.new; c = OpenStudio::Model::CurveLinear.new(m); c.evaluate(10)"
<BLANK>
```

**CLI: --verbose: it's shown (amongst a LOT of output that I'm truncating)**

```
$ openstudio --verbose -e "m = OpenStudio::Model::Model.new; c = OpenStudio::Model::CurveLinear.new(m); c.evaluate(10)"
[...]
D, [2021-04-28T14:42:32.271126 #29586] DEBUG -- : Activating gem :/ruby/2.5.0/specifications/ruby-ll-2.1.2.gemspec
D, [2021-04-28T14:42:32.271251 #29586] DEBUG -- : Activating gem :/ruby/2.5.0/specifications/oga-3.2.gemspec
D, [2021-04-28T14:42:32.271371 #29586] DEBUG -- : Activating gem :/ruby/2.5.0/specifications/openstudio-workflow-2.1.0.gemspec
D, [2021-04-28T14:42:32.271437 #29586] DEBUG -- : Activating gem :/ruby/2.5.0/specifications/openstudio-extension-0.3.1.gemspec
D, [2021-04-28T14:42:32.271540 #29586] DEBUG -- : Activating gem :/ruby/2.5.0/specifications/openstudio-standards-0.2.12.gemspec
D, [2021-04-28T14:42:32.271810 #29586] DEBUG -- : Executing cmd: m = OpenStudio::Model::Model.new; c = OpenStudio::Model::CurveLinear.new(m); c.evaluate(10)
[openstudio.model.CurveLinear] <0> Supplied x is above the maximumValueofx, resetting it.
```

### Environment

I tested this on Mac using OpenStudio 3.2alpha installer, but I don't think it is unique to this release, but I have not tried on Windows.

### Context

I wasn't aware of this limitation so I spent a lot of time looking through file by hand for problem. I know now that ruby is work around, but not all developers will have or want to use system Ruby, the CLI. I did notice our CI seems to show this log (which is How I found it). @tijcolem not sure why that worked because I thought it was using CLI to call measure tests. In my local test I just ran the measure tests and didn't make use of `openstudio measure run_tests` which could be the difference. I just called `openstudio --verbose my_test.rb`.

### PR DESCRIPTION
**Pull request by tijcolem**
Pull request overview
---------------------

 - Fixes #4309

Many users have a bash profile setup that automatically sets the RUBYLIB env to point to OpenStudio Ruby bindings (e.g. export RUBYLIB=/Applications/OpenStudio-3.2.1/Ruby), and while this works fine when using Ruby bindings, when using the cli directly odd runtime behavior happens such as changing the verbosity levels of log messages as described in the bug report. 

If a user wants to add ruby libs and modify the the ruby load path,  the options -`-include` or `-I` should be passed to the CLI vs setting this RUBYLIB.  This is what was done for bundler options such as GEM_HOME, GEM_PATH.  The main issue with these ruby specific envs is other embedded ruby gems can use these variables and cause some runtime issues, which  is what is happening with this bug. 

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/cli/embedded_help.rb [hunk 1]
```diff
@@ -54,9 +54,10 @@ def self.preprocess_ruby_script(s)
 module Kernel
   # ":" is our root path to the embedded file system
   # make sure it is in the ruby load path
-  if ENV['RUBYLIB']
-    ENV['RUBYLIB'].split(File::PATH_SEPARATOR).each {|lib| $LOAD_PATH.unshift(lib)}
-  end
+  # TJC remove RUBYLIB env and use --include or -I via cli args instead
+  #if ENV['RUBYLIB']
+  #  ENV['RUBYLIB'].split(File::PATH_SEPARATOR).each {|lib| $LOAD_PATH.unshift(lib)}
+  #end
```
Comment (jmarrec) [inline src/cli/embedded_help.rb:60]: I'm all good with these changes, but I think we need to document that properly **somewhere**, don't we? We should change the wording in https://nrel.github.io/OpenStudio-user-documentation/reference/command_line_interface/#environment-variables

**Review by jmarrec — APPROVED** [review]
Approved, if we do document it

File: src/cli/embedded_help.rb (hunk 1 above)
Comment (tijcolem) [reply src/cli/embedded_help.rb:60]: Yes, I'll take care of the documentation before I merge in this PR. We need to remove some of the wording in those docs related to using other ruby ENVs as well so I'll clean that up.

Comment (tijcolem) [reply src/cli/embedded_help.rb:60]: docs are updated to reflect this change

**Pull request merged**